web/src/games/bingo/board.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import { GameLayout } from 'gamesShared/components/fbg/GameLayout';
+import { isOnlineGame } from 'gamesShared/helpers/gameMode';
+import { Button } from '@material-ui/core';
+
+import { IBoardProps, IBoardState, INumberState } from './definitions';
+import {
+  GRID_SIZE,
+  CALL_BOX_SIZE,
+  MAX_BINGO_CALLS,
+  TIME_OUT,
+  INITIAL_WAIT_REF_NUM,
+  INITIAL_WAIT_TIME,
+  BACKOFF_INTERVAL,
+} from './constants';
+import PlayCard from './components/playCard';
+import CallCard from './components/callCard';
+import Countdown from './components/countDown';
+import CallTable from './components/callTable';
+
+export class BingoBoard extends React.Component<IBoardProps, IBoardState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showCallTable: false,
+    };
+  }
+
+  _getPlayerID = () => this.props.playerID || this.props.ctx.currentPlayer;
+
+  _callOnTimeout = (callRef: number) => {
+    if (callRef >= this.props.G.callRef) {
+      this.props.moves.incrementCallRef(this._getPlayerID());
+    }
+  };
+
+  _numberClicked = (number: INumberState) => {
+    this.props.moves.playerClickedNumber(number, this._getPlayerID());
+  };
+
+  _shoutBingo = () => {
+    this.props.moves.playerShouted(this._getPlayerID());
+  };
+
+  _gameOverStatus = () => {
+    if (this.props.ctx.gameover.draw) {
+      return 'draw';
+    }
+    if (isOnlineGame(this.props.gameArgs)) {
+      if (this.props.ctx.gameover.winner === this.props.playerID) {
+        return 'you won';
+      } else {
+        return 'you lost';
+      }
+    } else {
+      return `Player ${this.props.ctx.gameover.winner} won`;
+    }
+  };
+
+  _renderFooter = () => {
+    const btnStyles = {
+      marginTop: '10px',
+      alignContent: 'center',
+    };
+
+    const stars = new Array(MAX_BINGO_CALLS)
+      .fill('⭐')
+      .map((s, idx) => (this.props.G.players[this._getPlayerID()].shoutCount > idx ? s : '✰'));
+
+    return (
+      <div style={{ display: 'flex' }}>
+        <Button
+          key="bi_call_table_btn"
+          variant="contained"
+          color="primary"
+          style={{
+            ...btnStyles,
+            justifyContent: 'left',
+          }}
+          onClick={() => this.setState({ showCallTable: !this.state.showCallTable })}
+        >
+          Table
+        </Button>
+        <Button
+          key="bi_shout_btn"
```

Comment:
This function is very large. Can you move these buttons to their own functions, private to the class?

Reply:
Well they needed a common div, otherewise I had it seperate. But IMHO, they are just 2 buttons and also have common styles, so they can stay.